### PR TITLE
Drop `git` in gemspec

### PR DESCRIPTION
**Pull request by utkarsh2102**
Hi @iliabylich,

As discussed in #743, here's the PR, dropping `git` in gemspec.
Rationale at: https://docs.rubocop.org/rubocop-packaging/cops_packaging.html#gemspec-git-rationale

This helps in maintaining this in Debian and other downstream 💫 

Signed-off-by: Utkarsh Gupta <<utkarsh@debian.org>>

### CODE REVIEW

File: parser.gemspec [hunk 1]
```diff
@@ -20,27 +21,26 @@ Gem::Specification.new do |spec|
...
-                          lib/parser/ruby25.rb
-                          lib/parser/ruby26.rb
-                          lib/parser/ruby27.rb
-                          lib/parser/ruby30.rb
-                          lib/parser/macruby.rb
-                          lib/parser/rubymotion.rb
-                       ) - %w(
-                          .gitignore
-                       )
+  spec.files         = Rake::FileList['{bin,lib}/**/*'].exclude(*File.read('.gitignore').split)
```
Comment (iliabylich) [inline parser.gemspec:24]: This is a bit hard to follow. Also it forces `.gitignore` to have entries that are valid for `FileList` (this I why you had to change it I suppose)

Does it work if you do this?

```ruby
Dir['bin/*'] + Dir['lib/**/*.rb'] + 'parser.gemspec'
```

You can check it by doing

```
$ gem uninstall parser
$ gem uninstall ast
$ rake build
$ gem install pkg/parser-2.7.1.4.gem
$ ruby-parse -e '1 + 1'
```

Comment (iliabylich) [reply parser.gemspec:24]: also `spec/test_files` seems to be optional, and we don't want to include test files into release build, could you remove this assignment too, please?

Comment (utkarsh2102) [reply parser.gemspec:24]: Perfect, anything that's pure Ruby works for us! Just not `git` 😄 
I'll fix this right away! 😄

Comment (utkarsh2102) [reply parser.gemspec]: Does it work for you now?

File: parser.gemspec [hunk 2]
```diff
@@ -20,27 +20,24 @@ Gem::Specification.new do |spec|
...
-                          lib/parser/ruby26.rb
-                          lib/parser/ruby27.rb
-                          lib/parser/ruby30.rb
-                          lib/parser/macruby.rb
-                          lib/parser/rubymotion.rb
-                       ) - %w(
-                          .gitignore
-                       )
+  spec.files         = Dir['bin/*'] + Dir['lib/**/*.rb'] + 'parser.gemspec' + %w(
+                           lib/parser/lexer.rb
```
Comment (iliabylich) [inline parser.gemspec:24]: `lib/parser/lexer.rb` and `lib/parser/ruby*.rb` are twice in the list. The last part `+ %w( ... )` should be removed.

File: parser.gemspec [hunk 3]
```diff
@@ -59,4 +56,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'simplecov', '~> 0.15.1'
 
   spec.add_development_dependency 'gauntlet'
+  spec.add_development_dependency 'rubocop-packaging'
```
Comment (iliabylich) [inline parser.gemspec:59]: do we need it?

**Review by iliabylich — CHANGES_REQUESTED** [review]

File: parser.gemspec (hunk 2 above)
Comment (utkarsh2102) [reply parser.gemspec:24]: Eee, right. My bad.

File: parser.gemspec (hunk 3 above)
Comment (utkarsh2102) [reply parser.gemspec:59]: It helps to ensure that the library is easily maintainable downstream (not only Debian, but other distributions as well!).
The `Packaging` extension was explicitly created to help downstream maintainers without compromising the experience of upstream maintainers. 

Let me know what you think?

Comment (iliabylich) [reply parser.gemspec:43]: it's a dependency that `parser` doesn't need. could you remove it please? it doesn't affect development process and it's not even required anywhere. it will only slow down CI builds for no practical reason

Comment (utkarsh2102) [reply parser.gemspec]: fine, removed.

**Comment by iliabylich** [comment]
@utkarsh2102 my bad, should be `+ ['parser.gemspec']` (because we concatenate arrays)

**Comment by utkarsh2102** [comment]
> @utkarsh2102 my bad, should be `+ ['parser.gemspec']` (because we concatenate arrays)

I've taken all together at once now. Should be good to go.

**Review by iliabylich — APPROVED** [review]
👍

**Comment by iliabylich** [comment]
@utkarsh2102 Thanks!

**Comment by utkarsh2102** [comment]
Hi @iliabylich,
I dropped you a mail, could you take a look at it, whenever you have time? 😄

File: parser.gemspec [hunk 4]
```diff
@@ -20,27 +20,8 @@ Gem::Specification.new do |spec|
...
-                          lib/parser/ruby26.rb
-                          lib/parser/ruby27.rb
-                          lib/parser/ruby30.rb
-                          lib/parser/macruby.rb
-                          lib/parser/rubymotion.rb
-                       ) - %w(
-                          .gitignore
-                       )
-
+  spec.files         = Dir['bin/*', 'lib/**/*.rb', 'parser.gemspec']
```
Comment (mbj) [inline parser.gemspec:23]: BTW, and this is not a bad thing: before `parser` would ship anything which was not gitignored explicity (or in the whitelist) this means that the `parser` gem would not ship tests anymore. Which is IMO a good thing.

Comment (utkarsh2102) [reply parser.gemspec:23]: Yep, happy to have it this way!
